packages/m2m-gateway/test/integration/purposes/deletePurpose.test.ts

```diff
@@ -0,0 +1,51 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { unsafeBrandId } from "pagopa-interop-models";
+import {
+  getMockedApiPurpose,
+  getMockWithMetadata,
+} from "pagopa-interop-commons-test";
+import {
+  expectApiClientGetToHaveBeenCalledWith,
+  expectApiClientPostToHaveBeenCalledWith,
+  mockDeletionPollingResponse,
+  mockInteropBeClients,
+  purposeService,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import { getMockM2MAdminAppContext } from "../../mockUtils.js";
+
+describe("deletePurpose", () => {
+  const mockApiPurpose = getMockWithMetadata(getMockedApiPurpose());
+
+  const mockDeletePurpose = vi.fn();
+  const mockGetPurpose = vi.fn(mockDeletionPollingResponse(mockApiPurpose, 2));
+
+  mockInteropBeClients.purposeProcessClient = {
+    getPurpose: mockGetPurpose,
+    deletePurpose: mockDeletePurpose,
+  } as unknown as PagoPAInteropBeClients["purposeProcessClient"];
+
+  beforeEach(() => {
+    mockDeletePurpose.mockClear();
+    mockGetPurpose.mockClear();
+  });
+
+  it("Should succeed and perform API clients calls", async () => {
+    await purposeService.deletePurpose(
+      unsafeBrandId(mockApiPurpose.data.id),
+      getMockM2MAdminAppContext()
+    );
+
+    expectApiClientPostToHaveBeenCalledWith({
+      mockPost: mockInteropBeClients.purposeProcessClient.deletePurpose,
+      params: { id: mockApiPurpose.data.id },
+    });
+    expectApiClientGetToHaveBeenCalledWith({
+      mockGet: mockInteropBeClients.purposeProcessClient.getPurpose,
+      params: { id: mockApiPurpose.data.id },
+    });
+    expect(
+      mockInteropBeClients.purposeProcessClient.getPurpose
+    ).toHaveBeenCalledTimes(2);
+  });
+});
```

Comment:
Could you add a test for the `pollingMaxRetriesExceeded` as we do in other routes?